src/Components/Index/ServiceSelectionComponent.tsx

```diff
@@ -33,51 +32,48 @@ import {
 } from '@grafana/ui';
 
 import { SelectFieldButton } from '../Forms/SelectFieldButton';
-import { explorationDS, VAR_DATASOURCE, VAR_FILTERS } from 'services/shared';
-import { map, Observable, Unsubscribable } from 'rxjs';
+import { explorationDS, VAR_DATASOURCE } from 'services/shared';
+import { map, Observable } from 'rxjs';
 import { ByLabelRepeater } from 'Components/ByLabelRepeater';
 import { GrotError } from 'Components/GrotError';
 import { getLiveTailControl, getLokiDatasource } from 'services/scenes';
 import { getFavoriteServicesFromStorage } from 'services/store';
+import { debounce } from 'lodash';
 
 const LIMIT_SERVICES = 20;
 const SERVICE_NAME = 'service_name';
 
 interface ServiceSelectionComponentState extends SceneObjectState {
+  // The body of the component
   body: SceneCSSGridLayout;
-  repeater?: ByLabelRepeater;
-  groupBy: string;
-  metricFn: string;
-
-  showHeading?: boolean;
-  searchQuery?: string;
-  showPreviews?: boolean;
-  topServices?: string[];
-  isTopSeriesLoading: boolean;
+  // We query volume endpoint to get list of all services and order them by volume
+  listOfServicesOrderedByVolume?: string[];
+  // Keeps track of whether service list is being fetched from volume endpoint
+  isListOfServicesLoading: boolean;
+  // Keeps track of the search query in input field
   searchServicesString: string;
+  // List of services to be shown in the body
   topServicesToBeUsed?: string[];
 }
 
 export class ServiceSelectionComponent extends SceneObjectBase<ServiceSelectionComponentState> {
   protected _variableDependency = new VariableDependencyConfig(this, {
-    variableNames: [VAR_FILTERS, VAR_DATASOURCE],
+    // We want to subscribe to changes in datasource variables and update the top services when the datasource changes
+    variableNames: [VAR_DATASOURCE],
```

Comment:
I don't know why `VAR_FILTERS` was here - if anyone knows the reason, let me know, but all works as expected when I've removed it. 🤷‍♀️

Reply:
This is the only change I wouldn't be sure of the effect, because it defines "variable dependency", whatever that means, and whatever that internally does.

Reply:
I can bring it back and ask @adrapereira when he is back, if we want to be sure. Let me know. 🙂 But in this scene we are not depending on any other variable, just data source.

Reply:
🧡🧡🧡 Thank you for such a detailed explanation! Helped me understand it better. We are not using `VAR_FILTERS` so we don't need to do anything if they change - so it seems like it is okay to remove it. 🙂